app/models/concerns/hostext/search.rb

```diff
@@ -109,11 +109,36 @@ def search_by_user(key, operator, value)
         end
         key_name = User.connection.quote_column_name(clean_key)
         condition = sanitize_sql_for_conditions(["#{key_name} #{operator} ?", value_to_sql(operator, value)])
-        users = User.where(condition)
-        hosts = users.map(&:hosts).flatten
-        opts  = hosts.empty? ? "< 0" : "IN (#{hosts.map(&:id).join(',')})"
 
-        {:conditions => " hosts.id #{opts} " }
+        user_ids = User.where(condition).reorder(nil).pluck(:id)
+
+        return { conditions: '1 = 0' } if user_ids.empty?
+
+        usergroup_ids = Usergroup.joins(:cached_usergroup_members)
+                                 .where(cached_usergroup_members: {user_id: user_ids})
+                                 .reorder(nil)
+                                 .distinct
+                                 .pluck(:id)
```

Comment:
I think you can get the usergroup id directly from the CachedUsergroupMemeber table:
```
usergroup_ids = CachedUsergroupMember.where(user_id: user_id).pluck(:usergroup_id)
```

Reply:
Does this correctly check Taxonomies and Usergroup Permissions?

Reply:
Ok, Usergroups are not Taxable and we probably want to ignore permissions for this. Makes sense, I will apply the suggested change.